sippy-ng/src/prow_job_runs/ProwJobRun.js

```diff
@@ -0,0 +1,926 @@
+import * as lodash from 'lodash'
+import { Button, ButtonGroup, TextField } from '@material-ui/core'
+import { Error } from '@material-ui/icons'
+import { useHistory } from 'react-router-dom'
+import Alert from '@material-ui/lab/Alert'
+import PropTypes from 'prop-types'
+import React, { Fragment, useEffect, useState } from 'react'
+import TimelineChart from '../components/TimelineChart'
+
+export default function ProwJobRun(props) {
+  const history = useHistory()
+
+  const [fetchError, setFetchError] = React.useState('')
+  const [isLoaded, setLoaded] = React.useState(false)
+  const [eventIntervals, setEventIntervals] = React.useState([])
+  const [filteredIntervals, setFilteredIntervals] = React.useState([])
+
+  // TODO: Not sure this needs to be a useState param, really just need a constant map but
+  // wasn't working.
+  const [allCategories, setAllCategories] = useState(
+    new Map([
+      ['operator_unavailable', 'Operator Unavailable'],
+      ['operator_progressing', 'Operator Progressing'],
+      ['operator_degraded', 'Operator Degraded'],
+      ['pods', 'Pods (careful)'],
+      ['pod_logs', 'Pod Logs'],
+      ['interesting_events', 'Interesting Events'],
+      ['alerts', 'Alerts'],
+      ['node_state', 'Node State'],
+      ['e2e_test_failed', 'E2E Failed'],
+      ['e2e_test_flaked', 'E2E Flaked'],
+      ['e2e_test_passed', 'E2E Passed'],
+      ['disruption', 'Disruption'],
+    ])
+  )
+
+  const [categoryButtonState, setCategoryButtonState] = useState(() => {
+    const params = new URLSearchParams(window.location.search)
+
+    let categories = []
+    let categoriesParam = params.get('categories')
+    if (categoriesParam) {
+      categories = categoriesParam.split(',')
+    }
+
+    const initialState = {}
+    let atLeastOneCategorySelected = false
+    allCategories.forEach((value, key, map) => {
+      initialState[key] = categories.includes(key)
+      if (categories.includes(key)) {
+        atLeastOneCategorySelected = true
+      }
+    })
+
+    if (!atLeastOneCategorySelected) {
+      console.log('no category was selected, enabling the default set')
+      initialState['operator_unavailable'] = true
+      initialState['operator_progressing'] = true
+      initialState['operator_degraded'] = true
+      initialState['pod_logs'] = true
+      initialState['interesting_events'] = true
+      initialState['alerts'] = true
+      initialState['node_state'] = true
+      initialState['e2e_test_failed'] = true
+      initialState['disruption'] = true
+    }
```

Comment:
If you used useQueryParams, you wouldn't need any of this.  I'd expect this to be a default prop on the component like this:

```js
ProwJobRun.defaultProps = {
  categories:  {
    operator_unavailable: true,
    operator_progressing: true,
    operator_degraded: true,
    pod_logs: true,
    interesting_events: true,
    alerts: true,
    node_state: true,
    e2e_test_failed: true,
    disruption: true,
  },
}

ProwJobRun.propTypes = {
  categories: PropTypes.object,
}
```

And then for useQueryParams, the default value would be the prop:

```
 const [names = props.categories, setCategories] = useQueryParam('category', ArrayParam) 
```

Reply:
And then if someone clicked the button, you'd just call `setCategories(<new categories map>)` and the URL would be updated, the categories variable would get updated, and the intervals table should automatically rerender

Reply:
Gave it a shot, let me know what you think.